CMakeLists.txt

```diff
@@ -1,148 +1,34 @@
-cmake_minimum_required(VERSION 3.18)
+# Provide general project configuration setup
+#
+cmake_minimum_required(VERSION 3.21)
+
+set(STATUS_QT_VERSION 6.3)
+set(STATUS_VERSION 0.6.0)
 
 project(status-desktop
-  VERSION 0.1.0
+  VERSION ${STATUS_VERSION}
   LANGUAGES CXX)
 
-if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
-  message(
-    FATAL_ERROR
-      "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there."
-  )
-endif()
-
-set(CMAKE_INCLUDE_CURRENT_DIR ON)
-
-set(CMAKE_AUTOUIC ON)
-set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTORCC ON)
+set(PROJECT_ORGANIZATION_DOMAIN "status.im")
+set(PROJECT_ORGANIZATION_NAME "Status")
+set(PROJECT_APPLICATION_NAME "Status Desktop")
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
why 20? its supported for every platform?

Reply:
I checked some time ago, and most of the important features were supported on the leading platforms we target and the latest Android NDK. However, I'm not sure of the practical availability and the effort to build in CI using the supported compilers while keeping availability. This is a task on its own and will be tackled when the time is right.

Do you consider it to be safer to limit that availability to C++17 and bump it when we introduce c++20 features?